### PR TITLE
feat: add label to loop

### PR DESCRIPTION
**Pull request by XFFXFF**
closes: #12392 

I changed `'loop` to `'l`, as `cargo check` says `'loop` is an invalid label name.

### CODE REVIEW

File: crates/ide-assists/src/handlers/add_label_to_loop.rs [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        return None;
+    }
+    let loop_body = loop_expr.loop_body().and_then(|it| it.stmt_list());
+    let mut related_exprs = vec![];
+    related_exprs.push(ast::Expr::LoopExpr(loop_expr.clone()));
+    for_each_break_and_continue_expr(loop_expr.label(), loop_body, &mut |expr| {
+        if let ast::Expr::BreakExpr(_) | ast::Expr::ContinueExpr(_) = expr {
+            related_exprs.push(expr)
+        }
+    });
```
Comment (Veykril) [inline crates/ide-assists/src/handlers/add_label_to_loop.rs:39]: This can be moved into the `|builder|` closure for lazy evaluation

Comment (Veykril) [reply crates/ide-assists/src/handlers/add_label_to_loop.rs:39]: We also don't need to use a vec in that case then

Comment (XFFXFF) [reply crates/ide-assists/src/handlers/add_label_to_loop.rs:39]: Yes, it's better

File: crates/ide-assists/src/handlers/add_label_to_loop.rs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+// ```
+// fn main() {
+//     'l: loop {
+//         break 'l;
+//         continue 'l;
+//     }
+// }
+// ```
+pub(crate) fn add_label_to_loop(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let loop_expr = ctx.find_node_at_offset::<ast::LoopExpr>()?;
```
[the remark this answers is not in the document]
Comment (XFFXFF) [reply crates/ide-assists/src/handlers/add_label_to_loop.rs:28]: This makes more sense

**Comment by Veykril** [comment]
Thanks!
@bors r+

**Pull request merged**